subsys/shell/Kconfig

```diff
@@ -25,6 +25,19 @@ config SHELL_MINIMAL
 	  defaults which favor reduced flash or memory requirements over extra
 	  features.
 
+config SHELL_THREAD_PRIORITY_OVERRIDE
+	bool "Override default shell thread priority"
+	help
+	  Option to change the default value of shell thread priority.
```

Comment:
what is the default shell priority? Can you mention this here? so it is clear what we are overridng?

I am not following why we need 2 Kconfig variables to set something like this, seems way too complicated to me.

Reply:
default depends on whether SHELL_THREAD_PRIORITY_OVERRIDE is set yes or no.
- if it is set as no : then priority is governed by K_LOWEST_APPLICATION_THREAD_PRIO ( which depends on CONFIG_NUM_COOP_PRIORITIES or CONFIG_NUM_PREEMPT_PRIORITIES)

- if it is set as yes then  SHELL_THREAD_PRIORITY which has default 0.

the reason for 2 Kconfigs are following:
- I believe this is a limitation of Kconfig in general (not specific to Zephyr). There doesn't appear to be any support evaluating arithmetic expressions. we cant just directly give a default priority with a one Kconfig then it is failing this test in 
z_setup_new_thread()

#define Z_ASSERT_VALID_PRIO(prio, entry_point) do { \
	__ASSERT(Z_VALID_PRIO((prio), (entry_point)), \
		 "invalid priority (%d); allowed range: %d to %d", \
		 (prio), \
		 K_LOWEST_APPLICATION_THREAD_PRIO, \
		 K_HIGHEST_APPLICATION_THREAD_PRIO); \
	} while (false)

Reply:
Can you let me know next steps

Reply:
Can you let me know next steps.

Reply:
Can you let me know next steps?

Reply:
@udaykeetu - Kconfig does have a `range` feature, but I think it's limited to literals or other `CONFIG_` values.
@nashif - if there is a better solution, please consider making a separate PR.